src/i18n.ts

```diff
@@ -0,0 +1,22 @@
+import i18n from "i18next";
+import { initReactI18next } from "react-i18next";
+
+import Backend from "i18next-http-backend";
+import LanguageDetector from "i18next-browser-languagedetector";
+
+i18n
+  .use(Backend)
+  // detect user language
+  .use(LanguageDetector)
+  .use(initReactI18next)
+  .init({
+    fallbackLng: "zh",
+    interpolation: {
+      escapeValue: false, // not needed for react as it escapes by default
+    },
+    backend: {
+      loadPath: "/tshet-uinh-autoderiver/locales/{{lng}}/translation.json",
```

Comment:
This will need to follow `assetLocation` in `vite.config.ts` since we also have `vite.cos.config.ts` for production